lib/matplotlib/animation.py

```diff
@@ -1106,6 +1106,8 @@ def func(current_frame: int, total_frames: int) -> Any
                                      "save animations.")
                 _log.warning("MovieWriter %s unavailable; trying to use %s "
                              "instead.", writer, alt_writer)
+                if isinstance(alt_writer, str):
+                    alt_writer = writers[alt_writer]
```

Comment:
alt_writer is actually always a key in `writers`, so the `if` check could (should) be dropped.